kgl-glfw/src/nativeMain/kotlin/com/kgl/glfw/Window.kt

```diff
@@ -335,125 +335,152 @@ actual class Window @PublishedApi internal constructor(val ptr: CPointer<GLFWwin
 		}
 	}
 
-	actual fun setPosCallback(callback: WindowPosCallback?) {
+	actual fun setPosCallback(callback: WindowPosCallback?): WindowPosCallback? {
+		val previous = windowPosCallback
```

Comment:
We should have `setCallback` return the previous callback, to reduce the duplication.
Then for each one, we can just do `return setCallback(callback, { ......`.

Reply:
the native setCallback currently doesn't overwrite an existing callback with any new one (`current != null && new != null`), which is different behavior from both the JVM implementation in the same library, and the original C API. I'll go ahead and fix that while I'm making changes.

Reply:
The native `setCallback` doesn't overwrite the `staticCFunction` but it overwrites the property which is what is called in the end. I do that as an optimisation.

Reply:
ohhhh. that makes much more sense. I'll put that back in.

Reply:
The same optimization can be made for the JVM implementation. For full optimization, we might be able to set all the staticCFunctions (and JVM callbacks) at init, and then only switch out the callbacks.

Reply:
The full optimization would make the code simpler and less boilerplate-y.
My only concern is the trade-off between this and having callbacks that don't run for nothing.
Some of the callbacks (like cursor position iirc) can emit loads of events but don't conflate, so we'd have callbacks that don't do anything running.
I'm not sure how big a deal this is and it needs confirmation. Might be worth skipping for now?

Reply:
yeah. good point.